bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/editor/format/FormatOnSave.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   See git history
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.editor.format;
+
+import org.eclipse.cdt.lsp.editor.Configuration;
+import org.eclipse.cdt.lsp.editor.EditorOptions;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.lsp4e.format.IFormatRegionsProvider;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(property = { "serverDefinitionId:String=org.eclipse.cdt.lsp.server" })
+public class FormatOnSave implements IFormatRegionsProvider {
```

Comment:
With this new class reference we should the MANIFEST entry should be updated to a version that will include it - i.e.:

```
org.eclipse.lsp4e;bundle-version="0.17.2",
```

Reply:
I made a LSP4E PR to bump the version to 0.17.3. Then we can increase it here to 0.17.3 as well

Reply:
Unfortunately that means we can't release CDT LSP because it depends on an unreleased LSP4E. We can:

- Not merge this change and do release of 1.0.0 as is
- Remove the format on save changes from this PR, so that we are dependent on 0.17.1 of LSP4E
- Request and wait for LSP4E to release and then add the new version of LSP4E to our p2 site so it can be installed by users

WDYT?

Reply:
Option 4 is we do a soft/milestone "release" which can depend on snapshot LSP4E

Reply:
> Remove the format on save changes from this PR, so that we are dependent on 0.17.1 of LSP4E

I would prefer this one. I'll put the format-on-save stuff in a new PR.